tests/load/pipeline/test_redshift.py

```diff
@@ -41,3 +45,71 @@ def my_source() -> Any:
         assert pip_ex.value.__cause__.data_type == "time"
     else:
         assert pip_ex.value.__cause__.data_type in ("time", "binary")
+
+
+AWS_BUCKET = dlt.config.get("tests.bucket_url_s3", str)
+redshift_with_staging_configs = [
+    DestinationTestConfiguration(
+        destination_type="redshift",
+        staging="filesystem",
+        file_format="jsonl",
+        bucket_url=AWS_BUCKET,
+        extra_info="credential-forwarding",
+    ),
+    DestinationTestConfiguration(
+        destination_type="redshift",
+        staging="filesystem",
+        file_format="jsonl",
+        bucket_url=AWS_BUCKET,
+        staging_iam_role="arn:aws:iam::267388281016:role/redshift_s3_read",
+        extra_info="s3-role",
+    ),
+    ## for parquet, region should be unset
+    DestinationTestConfiguration(
+        destination_type="redshift",
+        staging="filesystem",
+        file_format="parquet",
+        bucket_url=AWS_BUCKET,
+        staging_iam_role="arn:aws:iam::267388281016:role/redshift_s3_read",
+        extra_info="s3-role",
+    ),
+]
+staging_regions = ["eu-central-1", ""]
```

Comment:
yeah a real test would require us to setup cross-region bucket and load data from there. I think can stick to the setup we have